mesonbuild/interpreterbase.py

```diff
@@ -449,6 +449,16 @@ def int_method_call(self, obj, method_name, args):
                 return obj % 2 != 0
             else:
                 raise InterpreterException('int.is_odd() must have no arguments.')
+        elif method_name == 'to_string':
+            if not posargs:
+                return str(obj)
+            elif len(posargs) == 1 and isinstance(posargs[0], str):
+                f = 'd' if len(posargs[0].strip()) == 0 else posargs[0]
+                if re.match('^[bcdoxX]$', f) is None:
+                    raise InvalidCode('Invalid format for int to string conversion "%s"' % f)
+                return str(('{:' + f + '}').format(obj))
```

Comment:
Why not use a kwarg for the base instead? Wouldn't that be simpler to parse and to use?

Reply:
I just followed [the approach](https://github.com/mesonbuild/meson/blob/master/mesonbuild/interpreterbase.py#L419) used by `bool`'s `to_string` method.

Although I don't have any preference, I just wanted to be consistent.